### ISSUE TITLE
CoverTransformer

### ISSUE THREAD

**Issue by remiadon** [scikit-mine/scikit-mine#96]
#### Describe the workflow you want to enable
Currently there is a confusing aspect to how pattern mining can serve Machine Learning workflows 
As suggested by @HugoPerrier Itemset mining algorithms (and their sequence counterpart ?) can be used as preprocessing steps to handle categorical data, and potentially replace [methods from sklearn](https://scikit-learn.org/stable/modules/preprocessing.html#encoding-categorical-features)

#### Describe your proposed solution
Add a CoverTransformer class that
- uses a BaseMiner instance, ensuring it has a `cover` function
- implements a `transform` method, calling `cover` on new data and returning the result in a tabular format
- set a `k` parameter, to lock dimensionality of the output (usage should be a good metric)

```python
>> from skmine.preprocessing import CoverTransformer; from skmine.itemsets import SLIM
>> slim_transformer = CoverTransformer(base_miner=SLIM(), k=2)
>> sllim_transformer.fit_transform([["banana", "milk", "bread", "orange"], ["banana", "milk"], ["bread"]])
   banana & milk  bread
0       0.800000    0.4
1       0.666667    0.0
2       0.000000    0.8

```

#### Describe alternatives you've considered, if relevant
A CoverTransformerMixin that can be inherited freely, like:
```python
>> from skmine.preprocessing import CoverTransformerMixin; from skmine.itemsets import SLIM
>> class SLIMTransformer(SLIM, CoverTransformerMixin): pass
...

```

**Comment by remiadon** [comment]
Also to be done : remove metion to `TransactionEncoder` in doc

**Comment by remiadon** [comment]
A SlimVectorizer has been added in v0.0.6 and mentions to the transaction Encoder have been removed
